ucsmsdk/ucsxmlcodec.py

```diff
@@ -82,7 +82,8 @@ def from_xml_str(xml_str, handle=None):
     """
 
     try:
-        root_elem = ET.fromstring(xml_str)
+        xml_raw_str = ucsgenutils.add_escape_chars(xml_str)
```

Comment:
Others  "invalid characters" ie '<', '>' and '&' are also modified in this function only. The difference being that these aree invalid characters and should not occur and hence checked in only during the exception in parsing. Whereas newline is valid character which must be escaped and hence needs to be in regular flow.